3rdparty/openssl/unsupported.h

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifndef _OE_OPENSSL_UNSUPPORTED_H
+#define _OE_OPENSSL_UNSUPPORTED_H
+
+#if !defined(OE_OPENSSL_SUPPRESS_UNSUPPORTED)
+
+#if defined(__cplusplus)
+#define OE_OPENSSL_EXTERN_C_BEGIN \
+    extern "C"                    \
+    {
+#define OE_OPENSSL_EXTERN_C_END }
+#else
+#define OE_OPENSSL_EXTERN_C_BEGIN
+#define OE_OPENSSL_EXTERN_C_END
+#endif
+
+#if defined(__GNUC__)
+#define OE_OPENSSL_UNSUPPORTED(MSG) __attribute__((deprecated(MSG)))
+#else
+#define OE_OPENSSL_UNSUPPORTED(MSG)
+#endif
+#define OE_OPENSSL_RAISE_ERROR(NAME) (OE_OPENSSL_UNSUPPORTED_ERROR, NAME)
```

Comment:
GCC now has a `error` attribute just like `deprecated` attribute. It will raise an error if only if the function is used.
clang does not yet support it. 
So our macro trick is good for now.